Pocket/pocket-ios
# feat(search): add overflow menu

## Summary
User can activate result overflow button

## References 
IN-1123

## Implementation Details
* Create new `OverflowMenu` view for overflow actions (add tags, archive/unarchive and delete). 
* Added `SavedItemsController` so that `SearchViewModel` can observe changes from core data
* Added 5 new search analytic actions

Note: Some issues may still exist and will be done in future tickets
* All Items have both Saves and Archive (archive and deleting is weird for this case) - I would like to discuss with the team on how to handle this and vote a new ticket to be created.
* Resubmitting search after performing an action will cause some inconsistencies in the search result view - ticket for edge case

## Test Steps
- [ ] GIVEN search is active
AND a search has been performed
AND there are 1+ results
AND search results are visible
THEN every search result includes an overflow button

- [ ] GIVEN that a user taps on the search result overflow button 
THEN the search results overflow menu is shown exactly as it would be if the same overflow button were tapped from Saved/Archived

**For All Items scope:**
1. If a user taps on add tags option, the user should be able to see the add tags view.
2. If a user deletes an item, the user should see that the item is deleted and removed the list.
3. If a user archives or moves an item to saves, the user should still see the item in the list, but update the cell to reflect “Archive” or “Move to Saves” depending on the type of item.

## PR Checklist:

## Screenshots
https://user-images.githubusercontent.com/6743397/221930660-3146e072-932c-45c6-92fa-f41ddb7c86ff.mp4
